File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.java
import java.util.*;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        
        Queue<Character> queue = new LinkedList<Character>();
    for(int i=0; i<skill_trees.length; i++){
      for(int k=0; k<skill.length(); k++){
        queue.add(skill.charAt(k));
      }

      if(check(skill_trees[i], queue)){
        answer++;
      }
      queue.clear();
    }
        return answer;
    }
    public boolean check(String skill, Queue<Character> queue){
    for(int i=0; i<skill.length(); i++){
      char target = skill.charAt(i);
      if(queue.contains(target)){
          Character q = queue.poll();
        if(!q.equals(target)) return false;
      }
    }

    return true;
  }
}

File: 프로그래머스/2/49993. 스킬트리/스킬트리.js
function solution(skill, skill_trees) {
    let answer = 0;
    
    for (const skillTrees of skill_trees) {
        let check = true;
        let index = 0;
        for (let i = 0; i < skillTrees.length; i++) {
            if (!skill.includes(skillTrees[i])) continue;
            if (skill[index] !== skillTrees[i]) {
                check = false;
                break;
            }
            index++;
        }
        
        if (check) answer++;
    }
    
    return answer;
}